utils/build_swift/postbuild_steps.py

```diff
@@ -0,0 +1,115 @@
+from swift_build_support.swift_build_support.targets \
+    import StdlibDeploymentTarget
+
+def create_symbols_package(args):
+    assert args.symbols_package
+    print('--- Creating symbols package ---')
+    print('-- Package file: {} --'.format(args.symbols_package))
+
+    if platform.system() == 'Darwin':
+        prefix = targets.darwin_toolchain_prefix(args.install_prefix)
+    else:
+        prefix = args.install_prefix
+
+    # As a security measure, `tar` normally strips leading '/' from paths
+    # it is archiving. To stay safe, we change working directories, then
+    # run `tar` without the leading '/' (we remove it ourselves to keep
+    # `tar` from emitting a warning).
+    with shell.pushd(args.install_symroot):
+        tar.tar(source=prefix.lstrip('/'),
+                destination=args.symbols_package)
+
+def build_benchmarks_linux(invocation):
+    # Perform some sanity checking.
+    args = invocation.args
+
+    # For Linux, we need a full install root.
+    assert args.install_destdir, "Can only benchmark if we compile a dest root?!"
+    # We also need to have built and installed Foundation into the destdir.
+    assert args.install_foundation, "Must build/install foundation to do benchmarks."
+
+    install_destdir = os.path.join(args.install_destdir,
+                                   args.install_prefix.lstrip('/'))
+
+    workspace = invocation.workspace
+    llvm = products.LLVM
+    swift = products.Swift
+    host_target = StdlibDeploymentTarget.host_target().name
+
+    swift_src_dir = workspace.source_dir(swift.product_source_name())
+    benchmark_src_dir = os.path.join(swift_src_dir, 'benchmark')
+    llvm_build_dir = workspace.build_dir(host_target,
+                                         llvm.product_source_name())
+    benchmark_build_dir = workspace.build_dir(host_target, "swiftbench")
+
+    # Clean the directory. Since this is an external build, we will not get
+    # proper dependencies from swift's cmake.
+    if os.path.isdir(benchmark_build_dir):
+        shutil.rmtree(benchmark_build_dir)
+    os.makedirs(benchmark_build_dir)
+
+    cmake = invocation.toolchain.cmake
+    # We use the just built clang
+    clang = os.path.join(llvm_build_dir, 'bin', 'clang')
+
+    swift_exec = os.path.join(install_destdir, 'bin', 'swiftc')
+    swift_lib_path = os.path.join(install_destdir, 'lib', 'swift')
+    num_o_iters = args.benchmark_num_o_iterations
+    num_onone_iters = args.benchmark_num_onone_iterations
+    configureInvocation = [
+        cmake,
+        '-GNinja',
+        '-DSWIFT_EXEC={}'.format(swift_exec),
+        '-DCLANG_EXEC={}'.format(clang),
+        '-DSWIFT_LIBRARY_PATH={}'.format(swift_lib_path),
+        '-DSWIFT_BENCHMARK_NUM_O_ITERATIONS={}'.format(num_o_iters),
+        '-DSWIFT_BENCHMARK_NUM_ONONE_ITERATIONS={}'.format(num_onone_iters),
+        '{}'.format(benchmark_src_dir)
+    ]
+
+    buildInvocation = [
+        cmake, '--build', benchmark_build_dir, '--',
```

Comment:
This sort of command I usually like to spread it out on more lines and group options with their arguments for better visual parsing. In this case something like:

```Python
build_invocation = [
    cmake,
    '--build', benchmark_build_dir,
    '--',
    'swift-benchmark-{}'.format(host_target),
]
```

These variable names would also make more sense as `configure_command`, `build_command`, `run_command`, etc... rather than an invocation.